caseworker/advice/templatetags/advice_tags.py

```
@@ -70,8 +70,16 @@ def group_advice(case):
             )
 
         _add_team_decisions(grouped_advice)
+    context["grouped_advice"] = grouped_advice
+    return context
+
 
-    return {"grouped_advice": grouped_advice}
+@register.filter
+def get_denial_reason_display_values(denial_reasons, denial_reasons_display):
+    if denial_reasons and denial_reasons_display:
+        return ", ".join([denial_reasons_display.get(item, item) for item in denial_reasons])
+    else:
+        return ""
```

sorry the `else` is not required. Below takes care of empty list also,
```
return ", ".join([denial_reasons_display[item] for item in denial_reasons])
```

Also instead of using `.get` it is intentional to access it by key because `denial_reasons` are seeded and we come across something that is not there then it should fail and we can investigate further instead of taking the default value.